src/main/cpp/sim/Simulator.cpp

```diff
@@ -50,6 +50,22 @@ Simulator::Simulator()
 
 void Simulator::SetTrackIndexCase(bool track_index_case) { m_track_index_case = track_index_case; }
 
+unsigned int Simulator::GetDate() const{
```

Comment:
Deze constructie met strings en integers om een datum voor te stellen misschien vervangen door een methode aan Calendar toe te voegen die zijn de interne boost::gregorian::date doorgeeft, en dan CheckPoint aanpassen zodat het met zo'n date werkt in plaats van een unsigned integer.